shared/src/ContentReactor.Shared/BlobRepository/BlobRepository.cs

```diff
@@ -1,132 +1,139 @@
-﻿using System;
+﻿using Azure;
+using Azure.Storage;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.Sas;
+using System;
 using System.Collections.Generic;
 using System.IO;
-using System.Linq;
 using System.Threading.Tasks;
-using Microsoft.WindowsAzure.Storage;
-using Microsoft.WindowsAzure.Storage.Blob;
 
 namespace ContentReactor.Shared.BlobRepository
 {
     public interface IBlobRepository
     {
-        CloudBlockBlob CreatePlaceholderBlob(string containerName, string folderName, string blobId);
-        Task DownloadBlobAsync(CloudBlockBlob blob, Stream stream);
-        Task<CloudBlockBlob> UploadBlobAsync(string containerName, string folderName, string blobId, Stream stream, string contentType);
-        Task<CloudBlockBlob> GetBlobAsync(string containerName, string folderName, string blobId, bool includeAttributes = false);
-        Task<IList<CloudBlockBlob>> ListBlobsInFolderAsync(string containerName, string folderName);
-        Task<bool> BlobExistsAsync(CloudBlockBlob blob);
-        Task DeleteBlobAsync(string containerName, string folderName, string blobId);
-        Task UpdateBlobMetadataAsync(CloudBlockBlob blob);
-        string GetSasTokenForBlob(CloudBlockBlob blob, SharedAccessBlobPolicy sasPolicy);
-        Task<byte[]> GetBlobBytesAsync(CloudBlockBlob blob);
+        BlockBlobClient CreatePlaceholderBlob(string containerName, string blobId);
+        Task DownloadBlobAsync(BlockBlobClient blob, Stream stream);
+        Task<BlockBlobClient> UploadBlobAsync(string containerName, string blobId, Stream stream);
+        Task<BlockBlobClient> GetBlobAsync(string containerName, string blobId);
+        Task<IList<BlobItem>> ListBlobsAsync(string containerName);
+        Task<Response<bool>> BlobExistsAsync(BlockBlobClient blob);
+        Task DeleteBlobAsync(string containerName, string blobId);
+        Task UpdateBlobMetadataAsync(BlockBlobClient blob, IDictionary<string, string> metaData);
+        string GetSasTokenForBlob(BlockBlobClient blob, BlobSasBuilder sasBuilder);
+        Task<byte[]> GetBlobBytesAsync(BlockBlobClient blob);
     }
 
     public class BlobRepository : IBlobRepository
     {
-        private static readonly CloudStorageAccount StorageAccount = CloudStorageAccount.Parse(Environment.GetEnvironmentVariable("BlobConnectionString"));
-        private static readonly CloudBlobClient BlobClient;
+        private static readonly BlobServiceClient BlobServiceClient;
```

Comment:
Is this good to set BlobServiceClient to static ? Original sample design such client as a static field.

Reply:
@jongio please review.

Reply:
I believe that the client is intended to support the scenario of being long-lived, so `static` would be fine.    I'll defer to @tg-msft as the authoritative voice, however.